x/distribution/keeper/msg_server.go

```diff
@@ -104,71 +105,85 @@ func (k msgServer) WithdrawValidatorCommission(goCtx context.Context, msg *types
+		return nil, errors.Wrapf(sdkerrors.ErrInvalidRequest, "cannot update base or bonus proposer reward because these are deprecated fields")
 	}
 
-	if (!req.Params.BaseProposerReward.IsNil() && !req.Params.BaseProposerReward.IsZero()) || //nolint:staticcheck // deprecated but kept for backwards compatibility
-		(!req.Params.BonusProposerReward.IsNil() && !req.Params.BonusProposerReward.IsZero()) { //nolint:staticcheck // deprecated but kept for backwards compatibility
-		return nil, errorsmod.Wrapf(errors.ErrInvalidRequest, "cannot update base or bonus proposer reward because these are deprecated fields")
+	if err := msg.Params.ValidateBasic(); err != nil {
+		return nil, err
 	}
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	if err := k.SetParams(ctx, req.Params); err != nil {
+	if err := k.SetParams(ctx, msg.Params); err != nil {
 		return nil, err
 	}
 
 	return &types.MsgUpdateParamsResponse{}, nil
 }
 
-func (k msgServer) CommunityPoolSpend(goCtx context.Context, req *types.MsgCommunityPoolSpend) (*types.MsgCommunityPoolSpendResponse, error) {
-	if k.authority != req.Authority {
-		return nil, errorsmod.Wrapf(govtypes.ErrInvalidSigner, "invalid authority; expected %s, got %s", k.authority, req.Authority)
+func (k msgServer) CommunityPoolSpend(goCtx context.Context, msg *types.MsgCommunityPoolSpend) (*types.MsgCommunityPoolSpendResponse, error) {
+	if err := k.validateAuthority(msg.Authority); err != nil {
+		return nil, err
 	}
 
-	ctx := sdk.UnwrapSDKContext(goCtx)
+	if err := validateAmount(msg.Amount); err != nil {
+		return nil, err
+	}
 
-	recipient, err := k.authKeeper.StringToBytes(req.Recipient)
+	recipient, err := k.authKeeper.StringToBytes(msg.Recipient)
 	if err != nil {
 		return nil, err
 	}
 
 	if k.bankKeeper.BlockedAddr(recipient) {
-		return nil, errorsmod.Wrapf(errors.ErrUnauthorized, "%s is not allowed to receive external funds", req.Recipient)
+		return nil, errors.Wrapf(sdkerrors.ErrUnauthorized, "%s is not allowed to receive external funds", msg.Recipient)
 	}
 
-	if err := k.DistributeFromFeePool(ctx, req.Amount, recipient); err != nil {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	if err := k.DistributeFromFeePool(ctx, msg.Amount, recipient); err != nil {
 		return nil, err
 	}
 
 	logger := k.Logger(ctx)
-	logger.Info("transferred from the community pool to recipient", "amount", req.Amount.String(), "recipient", req.Recipient)
+	logger.Info("transferred from the community pool to recipient", "amount", msg.Amount.String(), "recipient", msg.Recipient)
 
 	return &types.MsgCommunityPoolSpendResponse{}, nil
 }
 
 func (k msgServer) DepositValidatorRewardsPool(goCtx context.Context, req *types.MsgDepositValidatorRewardsPool) (*types.MsgDepositValidatorRewardsPoolResponse, error) {
-	ctx := sdk.UnwrapSDKContext(goCtx)
+	if err := k.validateAuthority(req.Authority); err != nil {
```

Comment:
The authority was not checked against the module authority :o

Reply:
I think the functionality is to be able to deposit validator rewards pool from not just authority, it can be any other depositer. The changes you made makes sense to me.
Not sure, if there is more underlying meaning here.

Reply:
I think the same. @zmanian, was there a reason for this name?
(https://github.com/cosmos/cosmos-sdk/pull/14147)